dhis-2/dhis-services/dhis-service-validation/src/main/java/org/hisp/dhis/outlierdetection/util/OutlierDetectionUtils.java

```diff
@@ -27,50 +27,86 @@
  */
 package org.hisp.dhis.outlierdetection.util;
 
-import java.util.Date;
+import static org.hisp.dhis.feedback.ErrorCode.E2208;
+import static org.hisp.dhis.feedback.ErrorCode.E7131;
+
+import java.sql.SQLException;
 import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
+import org.hisp.dhis.common.IllegalQueryException;
+import org.hisp.dhis.common.QueryRuntimeException;
 import org.hisp.dhis.commons.util.TextUtils;
+import org.hisp.dhis.feedback.ErrorCode;
 import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.springframework.dao.DataAccessResourceFailureException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.jdbc.BadSqlGrammarException;
 
 /**
  * @author Lars Helge Overland
  */
+@Slf4j
 public class OutlierDetectionUtils {
+
+  public static final String ERR_MSG_TABLE_NOT_EXISTING =
```

Comment:
Could we find a way to centralize these constants? They seem to be duplicates of existing ones...
We have the same in `AnalyticsUtils` class.

Reply:
well, yes, what about move Analytics Utils to the service core package?

Reply:
done